### ISSUE TITLE
Civilization Ideas

### ISSUE THREAD

**Issue by frodare** [ToroCraft/ToroQuest#97]
Ah, ok. I also have a suggestion about what happens when you become enemies to a civilisation: Curses and Assassins
 
Gustaberg, Aquarius, and Solaris would put a curse on you, while the others send an assassin. Assassins might spawn in villages of all civs, not just the one you enfuriated.
 
Terra: Sends a´ Witch Assassin, invisible and attacking you with potion arrows
 
Gustaberg: Air Curse: When you reach a heigth of 180, there is a 10% chance you get an levitation effect. The same for 190, 200, ... , 260
 
Lumens: Paladin Assassin, visible, attacking with a fire sword and clad in armor
 
Aquarius: Water Curse: You take damage when touching water
 
Solaris: Sun Curse: At noon, there is a 30% chan´ce to get hit by a fireball out of nowhere
 
Wolf: Night Assassin, Invisible, very fast, attacks using an Obsidian sword (might drop heavily damaged), blinds you on a full or new moon
 
I also have a pantheon prepared (from a while ago for a similar project that didn´t get past the planning stage), which might be useful to flesh this out. All civs use this pantheon, but think of different ones as the most important ones, reflecting their values. So here would be my idea of the main god of each civ:
 
Terra: Rovwelord, Genus Locii of the Earth and God of Stability, Strength and Perceverance
 
Gustaberg: The Cloud Chicken, Trickster God of Joy, Luck, and Prosperity
 
Lumens: Pyrron, the Destroyer-Creator, God of Change, Ambition, and Cooking
 
Aquarius: The Ancient Guardian, God of the Sea and the Rain, Wave-That-Sinks-The-Ship-Yet-Saves-The-Crew, Saviour-Traitor, The Salty One
 
Solaris: Sol Amans, The Loving Sun, God of Generosity, Love, and Wrathful Revenge
 
Wolf: Gray-Night, The Protecting Wolf, Silver Light in the Dark, God of Hospitality, and Protection from Danger. Also Blood-Night, God of the Vengeful Hunt
 
What do you think? Does that fit to what little you have already?

~CawCawDrakon

https://minecraft.curseforge.com/projects/toroquest?comment=20

**Comment by zeriley** [comment]
There are some great ideas in this.  I love the idea of assassins or even a bounty being placed on your head if you're enemies with a civilization.  We would have to change it so that once you were killed by the assassin or someone claimed your bounty, you would have to go back to neutral rep, so that it's not just an endless cycle of having assassins after you.  We could start with a simple assassin that wears armor in the color of the civilization they represent, and then evolve them into more diverse assassins later.

I'm not a big fan of the curses, unless they were very temporary like the elder guardian's mining fatigue.

The idea of deities for each civilization is intriguing, but most of the gameplay mechanics I can think of for them would be redundant to the village lord.  For instance, donating stuff to the god of Gustaberg would give you Fortune-like effects where no matter what pickaxe you use, you have a chance of getting double or triple drops.  Which we could either fold that idea into positive reputation for each civilization (positive rep with Gustaberg grants luck) or have two systems of reputation.  Reputation with Gustaberg allows you to buy weapons at a discount from the shopkeeper, reputation with The Cloud Chicken (love that name, by the way) would grant you more luck, but no reputation with the civilization.

That would give you a choice to make when donating any goods you may have.  Do you want reputation with the village or the deity?

Certainly some interesting ideas.  Most of them seem feasible, as well.  What do the rest of you think?

**Comment by frodare** [comment]
I like the idea of the curses, but yeah, they should be temporary.  I was thinking of maybe just hacking into the potion effect system.  Love the idea of a curse that would not let you touch water.  Maybe you get the curse when crossing into a province like a beacon, but the effects would last for a minute or so after leaving.

**Comment by frodare** [comment]
@zeriley you lost me on the reputation stuff, are you thinking of expanding the rep system beyond the 6 civs?

**Comment by zeriley** [comment]
Yeah, that's exactly what I had in mind.  Go through a village in a province and it'd be just like passing within so many blocks of an ocean temple.  How long does mining fatigue last?  Five minutes?  I guess it could be an effect that lasts proportional to the amount of negative rep.  Between -10 and -100, it would last for a minute; between -100 and -500, it would last for three minutes; under -500, it would last for five minutes.

**Comment by frodare** [comment]
I think we need a reason to have negative rep.  In all of our play testing, no one has ever wanted to go negative.  I think we need to figure that out first since there is not reason to all this cool stuff if no one would get it.

**Comment by zeriley** [comment]
I'm not thinking of doing anything with reputation at the moment.  Was just thinking through what gameplay mechanics could be added by having deities.  Most of the stuff I could think of could be accomplished by folding them into the village lord.

However, if we wanted to do deities, we could do rep for the civ and rep for the deity of that civ, separately.  So that it gives you a choice of how you want to be rewarded: cheaper and more access to goods or gameplay benefits like an ever-present fortune effect.

**Comment by zeriley** [comment]
Or we could just have deities just to give the village some more structures and more depth.  Could be purely cosmetic, but I'd rather it have some gameplay effect.

**Comment by frodare** [comment]
They could take the place of the village trophies.  The would take quite a bit of work though.

**Comment by zeriley** [comment]
Yeah, there's a lot of work in adding deities with gameplay mechanics.
 - the structure itself, which it sounds like CawCaw has already made
 - a new priest villager type
 - implementing the positive and negative rep (oooh, idea, you have negative rep with Cloud Chicken and sometimes diamonds turn into coal when mined)
 - quests the priest can give you (province quests are exclusive; if you have a quest from the village lord, you can't have a quest with the priest, and vice versa)
 - ways to lose rep with that deity

**Comment by CawCawDrakon** [comment]
Is there a way to add lorebooks for each civ? That was how I thought to implement stuff like religion, politics, history and so on. Maybe each civ gets their own librarian who can sell these books?
I would write them, but sadly my programming experience is only in R, so I can´t really help with the other stuff.

**Comment by zeriley** [comment]
That's why I think some enemy structures should be tied to civilizations.  When you clear out a Mage Tower, you would lose reputation with one of the civilizations.  So then you have to weigh the consequences of getting the loot from the mage tower or keeping your rep.

**Comment by frodare** [comment]
Certainly.  I am not sure how difficult it would be but I will look into to.  I was going to look into how vanilla books anyway for the quest system.  Would the lorebooks add to the game play?

**Comment by CawCawDrakon** [comment]
Maybe reading them for the first time gives a small rep bonus? Or they contain stuff that might be usefull - like pieces of an in-game wiki, to keep stuff immersive. 
Personally, I mainly play creative and turn survival for some artifact hunting & collecting, and since lorebooks count as artifacts, for me they would definitely add to the gameplay.

**Comment by zeriley** [comment]
Oooh, I like that.  One of the books could mention something about a nearby Bastion's Lair or a fabled deposit of diamonds.  If we can crack the nut for maps to mansions and monuments, a map could "fall out" of one of the lorebooks.

**Comment by CawCawDrakon** [comment]
About negative rep needing some reason: The assassins, if strong enough, count as minibosses. Some people would definitely invoke a civs wrath just to fight a new boss

**Comment by frodare** [comment]
We could have the rep boost, have them lead players to hidden artifacts in the map, have sets of books that give achievements when fully collected ... and much more :)

**Comment by zeriley** [comment]
Maybe if you die with one of the tools given to you by the lord or one of the lorebooks in your inventory, you'll lose rep for losing the civilization's stuff.  Maybe there are other ways of accidentally losing rep.

**Comment by CawCawDrakon** [comment]
I think it might be useful to have multiple lists of lorebooks, depending on where you can get them, like this:

1. Common Books
These you can get anywhere, general books, general game hints, lore that all civs share, etc.

2. - 7. Civ Books
Books that are unique to a civ. A book might be in multiple civ book lists. Show the views of each civ

8. Heretic Books
Rare and only found in dungeons. Quests from Booksellers to obtain them?

**Comment by zeriley** [comment]
@CawCawDrakon I think that's basically the idea we came up with above, with having different types of lorebooks.

Also, @frodare I've come back to the idea of turning the books into the village lord, because if you read the book outside of a province, what civ do you get rep for?  If it's a civilization's book, it would make sense to give that rep to the appropriate civilization, but for generic ones, where would it go?  If we have to turn them in to the village lord, then the rep would be given at that point, and that donation system is already built.

From there, maybe the village lord can have a bibliography tab in his GUI where you can re-read any books you have donated to that civilization (not province).

Sidenote: if you donate one civ's book to another civ, maybe you can lose rep with the intended civ and gain boosted rep from the civ donated to.

**Comment by CawCawDrakon** [comment]
Wouldn´t it make more sense to have another npc for these book donations? An archivist or something, with the same donation system as the village lord, just with books. Or you donate the books to the village lord and the archivist has the bibliography tab

**Comment by zeriley** [comment]
It may make more sense, but we can't get too ambitious.  We all have full-time jobs as programmers and it's important to start small or else features will never be finished.  Creating another entity with another GUI with more AI and a building for him to live in, potentially, creates a lot more work for us.  Start small and enhance down the road is the path to success for us.

**Comment by CawCawDrakon** [comment]
Ah, that makes sense.

**Comment by zeriley** [comment]
In light of that, please don't get discouraged if our first versions of the features you have asked for don't live up to your expectations.  This is an iterative process and we will slowly enhance each feature of the mod over time.

**Comment by CawCawDrakon** [comment]
Of course. But looking at the what you have already, I doubt it will dissapoint. Take your time and keep up the great work, etc, etc

**Comment by frodare** [comment]
My vote is to make the lorebooks craftable into into a complete set (so maybe 9 instead of 12) which could be placed as a block.  At first, all this would do is give you rep, xp and a collector's item (like the ender dragon egg).  Maybe we could add something if place all the book sets together.  Zane beat me to it, this might be mostly due to me being lazy ;)

**Comment by zeriley** [comment]
But what about the non-civilization lore books?

**Comment by frodare** [comment]
They would just be there for interest and hits / in game wiki replacement.  Maybe in the future they could even have quests unrelated to a civ.

**Comment by CawCawDrakon** [comment]
Apropo "Unrelated to a civ": Is there an easy way to implement nomads or neutral trading caravans, moving through the world, from village to village? Alternatively, those could spawn around some kind of way-stone structure in the wilderness as well as in villages, just implying travel? What do you think?

**Comment by frodare** [comment]
@CawCawDrakon that is getting back to some of our original ideas! I think it is possible so some degree.  An easier place to start might be giving a village a one in six chance to spawn neutral with no added structures.  We could then add a chance of spawning caravan mobs (with lamas of course:) on lines between known provinces and neutral villages. For that matter, we could spawn them between provinces.

**Comment by zeriley** [comment]
Oooooh, and some of those nomads won't trade with you if you have positive rep with certain civilizations, and they'll have really good items or good deals or something.  That gives us a reason to go negative.

**Comment by CawCawDrakon** [comment]
@zeriley This would also fit the lore I have written so far about Aquarius and Lumens engaging in slave-trading (unless you don´t want that)! So Aquarius and Lumens (and maybe one or two of the others) are enemies to the nomads, while Wolf is an ally to the nomads (because of their hospitality - in the lore, Wolf actually would be the only civ to let the nomads set up camp within their city walls. Impossible to implement (or at the very least not worth the huge effort), just a tidbit of lore)

**Comment by zeriley** [comment]
Ah, that's cool.  I have no problem with slave trading.  I mean, in the game... obviously, personally, I have many problems with slave trading.

**Comment by CawCawDrakon** [comment]
Another thing about the non-civ lorebooks: these gods are not the only ones in that Pantheon. So, if the civ-books Open a Portal their Main deities, the others might Send you to other gods.

Also, I don't think bookshelves are Worth the trouble: Bibliocraft will Upgrade eventually, and anyone interested in collecting books will have that

**Comment by Hypsandar** [comment]
Frodare, you post really great suggestions! I'd love to see the in the mod.
